### PR TITLE
Provide bindings for TabBar and TabItem

### PR DESCRIPTION
**Pull request by nazzrim**
Dear imGui v.1.67 introduces TabBar and TabItem APIs but the wrappers were missing so I created them.

### CODE REVIEW

File: Dir.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	// Left left direction provided
+	Left = 0
+	// Right right direction provided
+	Right = 1
+	// Up up direction provided
+	Up = 2
+	// Down down direction provided
+	Down = 3
+	// COUNT number of directions
+	COUNT
```
Comment (dertseha) [inline Dir.go]: Would this `COUNT` be used in any way?
I'd remove it, because it is named too generic and doesn't have any benefit.

File: Dir.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package imgui
+
+const (
+	// None no direction provided
+	None = -1
```
Comment (dertseha) [inline Dir.go]: `None` is also a little too generic, I propose `NoDirection`.

File: TabBarFlags.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	TabBarFlagsNone = 0
+	// TabBarFlagsReorderable Allow manually dragging tabs to re-order them + New
+	// tabs are appended at the end of list
+	TabBarFlagsReorderable = 1 << 0
+	// TabBarFlagsAutoSelectNewTabs Automatically select new tabs when they appear
+	TabBarFlagsAutoSelectNewTabs = 1 << 1
+	// TabBarFlagsNoCloseWithMiddleMouseButton Disable behavior of closing tabs (that are submitted with p_open != NULL) with middle mouse button. You can
+	// still repro this behavior on user's side with if (IsItemHovered() && IsMouseClicked(2)) *p_open = false.
+	TabBarFlagsNoCloseWithMiddleMouseButton = 1 << 2
+	// TabBarFlagsNoTabListPopupButton Undocumented in Dear imGui v1.67
```
Comment (dertseha) [inline TabBarFlags.go]: Although it is undocumented, it probably has a function.
Please, for this, and similar that don't have any information in the .h file, add information what they would do.

File: imguiWrapper.h [hunk 4]
```diff
@@ -7,61 +7,211 @@ extern "C"
...
 extern IggContext iggCreateContext(IggFontAtlas sharedFontAtlas);
 extern void iggDestroyContext(IggContext context);
 extern IggContext iggGetCurrentContext();
 extern void iggSetCurrentContext(IggContext context);
+//DebugCheckVersionAndDataLayout
 
-extern IggIO iggGetCurrentIO(void);
-extern IggGuiStyle iggGetCurrentStyle(void);
+/*************************/
+/* Main - v1.67 complete */
```
Comment (dertseha) [inline imguiWrapper.h]: I'd avoid adding concrete version numbers, as they have a high potential of becoming invalid over time.

File: imguiWrapper.h [hunk 5]
```diff
@@ -7,61 +7,211 @@ extern "C"
...
+extern IggVec2 iggGetCursorStartPos(void);
+extern IggVec2 iggGetCursorScreenPos(void);
+extern void iggSetCursorScreenPos(IggVec2 const *absPos);
+extern void iggAlignTextToFramePadding();
+extern float iggGetTextLineHeight(void);
+extern float iggGetTextLineHeightWithSpacing(void);
+// GetFrameHeight
+// GetFrameHeightWithSpacing
+
+// here because we don't want to break with previous versions of this wrapper
```
Comment (dertseha) [inline imguiWrapper.h]: I don't understand - what is the intent here? These functions are internal, so, they could be removed should they no longer be called from the Go side.

**Comment by nazzrim** [comment]
@dertseha I realized too late that this pull request references the head of master of my repository instead of commit d557b29.
Commit d557b29 adds bindings for TabItem and TabBar from Dear ImGui v1.67.
The later commits intoduce most of the points you mention. I wanted to provide bindings for the missing functions from Dear ImGui v1.67 and the repository contains a intermediate result.
But I abandoned that idea for now since I wanted to try cimgui for bindings of a recent Dear ImGui version.

**Comment by nazzrim** [comment]
Closed because the wrong branch is used. Sorry for that.

**Pull request closed without merging**